src/sentry/api/endpoints/project_key_details.py

```diff
@@ -50,74 +91,74 @@ def put(self, request: Request, project, key_id) -> Response:
         serializer = ProjectKeyRequestSerializer(data=request.data, partial=True)
         default_version = get_default_sdk_version_for_project(project)
 
-        if serializer.is_valid():
-            result = serializer.validated_data
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        result = serializer.validated_data
 
-            if result.get("name"):
-                key.label = result["name"]
-
-            if not key.data:
-                key.data = {}
+        if result.get("name"):
+            key.label = result["name"]
+        if not key.data:
+            key.data = {}
+        key.data["browserSdkVersion"] = (
+            default_version if not result.get("browserSdkVersion") else result["browserSdkVersion"]
+        )
 
-            key.data["browserSdkVersion"] = (
-                default_version
-                if not result.get("browserSdkVersion")
-                else result["browserSdkVersion"]
-            )
+        result_dynamic_sdk_options = result.get("dynamicSdkLoaderOptions")
+        if result_dynamic_sdk_options:
+            if key.data.get("dynamicSdkLoaderOptions"):
+                key.data["dynamicSdkLoaderOptions"].update(result_dynamic_sdk_options)
+            else:
+                key.data["dynamicSdkLoaderOptions"] = result_dynamic_sdk_options
+
+        if result.get("isActive") is True:
```

Comment:
Can we drop the `is True` and `is False`?

Reply:
Hmm I thought about this logic and at first it looks messy, but I actually think I prefer to keep it this way. `isActive` may not be set in which case we want to keep the same status. We can remove the `is True` easily, but the second elif has to be something like `elif not result.get("isActive", True):` to keep the same behavior which is hard to read and understand. I like keeping the top the same so it matches the bottom `elif` too

Reply:
you could set a default like `result.get("isActive", False)` so it defaults to `False`, otherwise it defaults to `None`

Reply:
although it might be that you might not have the field set at all so your thinking makes sense too

Reply:
Fun convo. Let me join in too 😛  My thought is that the current version of code has no indication that isActive might not be set at all so it's dangerous for next engineers modifying this code. Anything with setting the default value would work.